kedro_pandera/framework/hooks/pandera_hook.py

```diff
@@ -71,5 +73,20 @@ def before_node_run(  # noqa : PLR0913
                     f"(kedro-pandera) Dataset '{name}' was successfully validated with pandera"
                 )
 
+    @hook_impl
+    def before_node_run(  # noqa : PLR0913
+        self,
+        node: Node,
+        catalog: DataCatalog,
+        inputs: Dict[str, Any],
+        is_async,
+        session_id,
+    ):
+        return self._validate_datasets(node, catalog, inputs)
+
+    @hook_impl
+    def after_node_run(self, node: Node, catalog: DataCatalog, outputs: Dict[str, Any]):
+        return self._validate_datasets(node, catalog, outputs)
```

Comment:
- Indeed we should not return anything. Only when we also add the converting feature
- I added your suggestion. Although I think it would be nice if we make it configurable for example in the catalog.yaml
example:
```
"dataset":
    type: pandas.CSVDataset
    filepath: data/01_raw/data. csv
    metadata:
        pandera:
            schema: ${pa.yaml:_data_schema}
            before_only: True # validates the dataset only before the node run 
```

Reply:
Btw. would it be an Idea to use the hooks `after_dataset_loaded` for input validation and `before_dataset_saved` for output validations.